### PR TITLE
[MEDIA] Image croping

### PR DESCRIPTION
**Pull request by mihaifaget**
Related issue: https://github.com/backbee/backbee-standard/issues/666

### CODE REVIEW

File: Config/route.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+bundle.crop.docrop:
+    pattern: /rest/{version}/crop
+    defaults:
+        _action: cropAction
+        _controller: crop.controller
+    requirements:
+        version: \d
+        _method: POST
+        #filename: "[a-zA-Z_\/]+"
```
Comment (eric-chau) [inline Config/route.yml]: Remove this lane if you don't need it.

File: Controller/CropController.php [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+	BackBee\Exception\BBException,
+	BackBee\Utils\File\File,
+	BackBee\Rest\Controller\ClassContentController,
+	BackBee\Rest\Controller\Annotations as Rest,
+	BackBee\Rest\Controller\MediaController,
+	BackBee\ClassContent\AbstractClassContent,
+	BackBee\ClassContent\Revision,
+	Symfony\Component\HttpFoundation\JsonResponse,
+	Symfony\Component\HttpFoundation\Response,
+	Symfony\Component\HttpFoundation\Request;
```
Comment (eric-chau) [inline Controller/CropController.php]: Write one use statement per line instead of a grouped one please.

File: Controller/CropController.php [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+use BackBee\ClassContent\Repository\ClassContentRepository;
+/**
+ * CropBundle main class
+ *
+ * @category    BackBee
+ * @package     BackBee\Bundle\CropBundle
+ * @copyright   Lp digital system
+ * @author      Mihai Faget <mihai.faget@lp-digital.fr>
+ */
+class CropController extends ClassContentRepository
```
Comment (eric-chau) [inline Controller/CropController.php]: A controller must not extend a Repository! Always keep in mind the separation of concerns.

File: Controller/CropController.php [hunk 4]
```diff
@@ -0,0 +1,410 @@
...
+     */
+    public function __construct(BBApplication $application, Config $config, RouteCollection $routeCollection)
+    {
+        $this->application = $application;
+        $this->config = $config;
+        $this->routeCollection = $routeCollection;
+        $this->_em = $application->getEntityManager();
+        $this->revisionRepository = $this->_em->getRepository('BackBee\ClassContent\Revision');
+        $this->mediaRepository = $this->_em->getRepository('BackBee\NestedNode\Media');
+        $this->mediaDir = $application->getMediaDir();
```
Comment (eric-chau) [inline Controller/CropController.php]: You need to declare every attribute you use in the class definition for more readability.

File: Controller/CropController.php [hunk 5]
```diff
@@ -0,0 +1,410 @@
...
+    }
+
+    /**
+     * Image crop action
+	 * @Rest\Security("is_fully_authenticated() & has_role('ROLE_API_USER')")
+	 */
+    public function cropAction(Request $request)
```
Comment (eric-chau) [inline Controller/CropController.php]: Your docblock indentation is broken.

**Comment by eric-chau** [comment]
You have to follow PSR 1 and PSR 2 to keep the code consistent:

http://www.php-fig.org/psr/psr-1/
http://www.php-fig.org/psr/psr-2/

Thank you.

**Comment by mihaifaget** [comment]
files updated

**Pull request closed without merging**